### PR TITLE
Ik 115[Refactor] Added date and time constraints to the matching process

### PR DESCRIPTION
**Pull request by anrchen**
Adding date and time constraints to the matching process

### CODE REVIEW

File: client/app/src/main/java/DirectionModel/Matcher.java [hunk 1]
```diff
@@ -81,33 +86,41 @@ public void matchRoute(List<LatLng> routeOfUser) {
...
-                    if (this.validateDistance(drop, pointInPoly) && goToEnd) {
-                        for ( int key : matchedMarkers.keySet()) {
-                            if(key == passengerRouteId) {
-                                Marker marker = matchedMarkers.get(key);
-                                marker.setIcon(BitmapDescriptorFactory.defaultMarker(BitmapDescriptorFactory.HUE_ROSE));
-                                pickUpBool = true;
-                                break;
-                            }
-                        }
+    private void matchDistance(int i, int passengerRouteId, int dateMatched, List<LatLng> routeOfUser,
```
Comment (arosser) [inline client/app/src/main/java/DirectionModel/Matcher.java]: need to add method comment

File: client/app/src/main/java/DirectionModel/RouteDeserializer.java [hunk 2]
```diff
@@ -15,28 +15,27 @@
...
      */
-    public List<Route> parseJSON(String res) throws JSONException {
+    public Route parseJSON(String res) throws JSONException {
 
-        List<Route> routes = new ArrayList<Route>();
+        Route route = new Route();
         if (res == null)
-            return routes;
-
+            return route;
```
Comment (arosser) [inline client/app/src/main/java/DirectionModel/RouteDeserializer.java]: not indented correctly

File: client/app/src/main/java/DirectionModel/Matcher.java [hunk 3]
```diff
@@ -81,33 +86,41 @@ public void matchRoute(List<LatLng> routeOfUser) {
...
+                               boolean pickUpBool, boolean goToEnd, LatLng pickUp, LatLng drop){
+        while (i < routeOfUser.size() && !pickUpBool) {
+            LatLng pointInPoly = routeOfUser.get(i);
+            if (this.validateDistance(pickUp, pointInPoly) && !goToEnd && dateMatched==0) {
+                goToEnd = true;
+                i++;
+            }
+
+            if (this.validateDistance(drop, pointInPoly) && goToEnd && dateMatched==0) {
+                for ( int key : matchedMarkers.keySet()) {
```
Comment (adrielFab) [inline client/app/src/main/java/DirectionModel/Matcher.java]: can validateDistance be applied to only 1 condition?

**Review by arosser — CHANGES_REQUESTED** [review]

File: client/app/src/main/java/com/example/mrides/Activity/CreateRouteActivity.java [hunk 4]
```diff
@@ -317,63 +333,73 @@ public void startObtainDirection() {
...
+            polylineOptions.add(route.getPoints().get(i));
 
-            for (int i = 0; i < route.getPoints().size(); i++)
-                polylineOptions.add(route.getPoints().get(i));
+        polylinePaths.add(mGoogleMap.addPolyline(polylineOptions));
 
-            polylinePaths.add(mGoogleMap.addPolyline(polylineOptions));
+        DateFormat sdf = new SimpleDateFormat("mm-dd-yyyy hh:mm");
+        Date date = null;
+        String str_date = this.in_date + " " + this.in_time;
```
Comment (adrielFab) [inline client/app/src/main/java/com/example/mrides/Activity/CreateRouteActivity.java]: camelCase?

**Review by adrielFab — CHANGES_REQUESTED** [review]

**Review by arosser — APPROVED** [review]

**Review by adrielFab — APPROVED** [review]

**Pull request merged**